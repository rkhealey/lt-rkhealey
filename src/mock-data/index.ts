import profile from './profile.png';

export const user = {
  username: '@yourname',
  imageSrc: profile,
};

export const theme = {
  background_color: 'rebeccapurple',
  text_color: 'palegoldenrod',
};

export const links = [
  {
    url: 'http://www.google.com',
    text: 'Link One',
  },
  {
    url: 'http://www.google.com',
    text: 'Link Two',
  },
  {
    url: 'http://www.google.com',
    text: 'Link Three',
  },
  {
    url: 'http://www.google.com',
    text: 'Link Four',
  },
  {
    url: 'http://www.google.com',
    text: 'Link Five',
  },
];

export const showLinks = {
  title: 'Shows',
  links: [
    {
      date: 'April 01 2019',
      venue: 'The Forum, Melbourne',
      url: 'http://www.songkick.com/1',
      available: true,
    },
    {
      date: 'April 02 2019',
      venue: 'Venue Name, Canberra',
      url: 'http://www.songkick.com/2',
      available: false,
    },
    {
      date: 'April 03 2019',
      venue: 'Venue Name, Sydney',
      url: 'http://www.songkick.com/3',
      available: true,
    },
    {
      date: 'April 04 2019',
      venue: 'Venue Name, Brisbane',
      url: 'http://www.songkick.com/4',
      available: true,
    },
  ],
};
