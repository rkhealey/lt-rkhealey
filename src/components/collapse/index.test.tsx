import React from 'react';
import { shallow } from 'enzyme';
import { Button } from 'rebass/styled-components';
import Collapse from '.';

import { theme } from '../../mock-data';

describe('Collapse', () => {
  it('Renders', () => {
    const wrapper = shallow(
      <Collapse theme={theme} title="Foo">
        Content
      </Collapse>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Hides and shows the content', () => {
    const wrapper = shallow(
      <Collapse theme={theme} title="Foo">
        <p className="the-content">Content</p>
      </Collapse>,
    );
    expect(wrapper.find('.the-content')).toHaveLength(0);
    wrapper.find(Button).simulate('click');
    expect(wrapper.find('.the-content')).toHaveLength(1);
  });
});
