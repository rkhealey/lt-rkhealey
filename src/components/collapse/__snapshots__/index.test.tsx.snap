// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse Renders 1`] = `
<styled.div
  mb={3}
>
  <ForwardRef
    onClick={[Function]}
    sx={
      Object {
        ":hover": Object {
          "backgroundColor": "palegoldenrod",
          "color": "rebeccapurple",
        },
        "backgroundColor": "rebeccapurple",
        "color": "palegoldenrod",
      }
    }
    variant="block"
  >
    Foo
  </ForwardRef>
</styled.div>
`;
